### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by DenisPryt**
Fix some warnings and crashes: bracers in defines (out_geojson.c), pass a correct parameters to fwrite, Bit_Chain initialization, read section name according to name_length, free section name, check num_pages for zero (avoiding malloc(0)), fix posible memory leeks in dwg_api.c and reedsolomon.c

### CODE REVIEW

File: src/decode.c [hunk 1]
```diff
@@ -146,7 +146,7 @@ dwg_decode(Bit_Chain * dat, Dwg_Data * dwg)
   memset(&dwg->header_vars, 0, sizeof(Dwg_Header_Variables));
   memset(&dwg->r2004_header.file_ID_string[0], 0, sizeof(dwg->r2004_header));
   memset(&dwg->auxheader.aux_intro[0], 0, sizeof(dwg->auxheader));
-  memset(&dwg->second_header.size, 0, sizeof(dwg->second_header));
+  memset(&dwg->second_header, 0, sizeof(dwg->second_header));
```
Comment (rurban) [inline src/decode.c]: I remember vaguely there was a reason to only clear the size. Need to check

File: src/decode.c [hunk 2]
```diff
@@ -3016,7 +3015,7 @@ get_base_value_type(short gc)
               if (gc <= 469) return VT_REAL;
               if (gc <= 479) return VT_STRING;
               if (gc <= 998) return VT_INVALID;
-              if (gc == 999) return VT_STRING;
+              /*if (gc == 999)*/ return VT_STRING;
```
Comment (rurban) [inline src/decode.c]: why?

File: src/decode.c [hunk 3]
```diff
@@ -3027,15 +3026,15 @@ get_base_value_type(short gc)
               if (gc <= 409) return VT_INT16;
               if (gc <= 419) return VT_STRING;
               if (gc <= 429) return VT_INT32;
-              if (gc <= 439) return VT_STRING;
+              /*if (gc <= 439)*/ return VT_STRING;
```
Comment (rurban) [inline src/decode.c]: why? doesn't look right

File: src/decode_r2007.c [hunk 4]
```diff
@@ -157,7 +157,7 @@ static void copy_bytes(char *dst, uint32_t length, uint32_t offset);
 static uint32_t read_literal_length(unsigned char **src, unsigned char opcode);
 static void copy_compressed_bytes(char *restrict dst, char *restrict src, int length);
 static void  bfr_read(void *restrict dst, char **restrict src, size_t size);
-static DWGCHAR* bfr_read_string(char **src);
+static DWGCHAR* bfr_read_string(char **src, int64_t stringSize);
```
Comment (rurban) [inline src/decode_r2007.c]: good. it's static so it needs no version bump. 0.6.3 is fine

File: src/logging.h [hunk 5]
```diff
@@ -91,7 +91,7 @@
   if (DWG_LOGLEVEL >= DWG_LOGLEVEL_##level && wstr) { \
     BITCODE_TU ws = wstr;                             \
     uint16_t _c;                                      \
-    while ((_c = *ws++)) {                            \
+    while (ws && (_c = *ws++)) {                      \
```
Comment (rurban) [inline src/logging.h]: Not needed. We already checked wstr in line 91

Comment (DenisPryt) [reply src/logging.h]: Yor are right. I'll remove this.

File: src/decode.c (hunk 1 above)
Comment (DenisPryt) [reply src/decode.c]: size is the first field of the structure and sizeof all header. 
I think `&dwg->second_header.size` and `&dwg->second_header are equals`.
Maybe correct code is `memset(&dwg->second_header.size, 0, sizeof(dwg->second_header.size));`?

File: src/decode.c (hunk 3 above)
Comment (DenisPryt) [reply src/decode.c]: This not a mistake, just unnecessary `if`. This comparison is always true

File: src/decode.c (hunk 2 above)
Comment (DenisPryt) [reply src/decode.c]: Not a mistake. Just this comparison is always true

**Comment by DenisPryt** [comment]
I'm waiting for recommendations on `get_base_value_type` and `decode.c`

File: src/decode.c (hunk 1 above)
Comment (rurban) [reply src/decode.c]: Oh, good call. I didn't see that.

File: src/decode.c (hunk 3 above)
Comment (rurban) [reply src/decode.c]: I see. But I rather want to keep it in nevertheless then, please.

File: src/decode.c (hunk 1 above)
Comment (rurban) [reply src/decode.c]: Your patch is fine then

**Comment by rurban** [comment]
Merged with e254c3b2d91370c32dd6e1b50b109c6aee6437d4

**Pull request closed without merging**